File: Project/src/main/java/com/Techprimelab/Project/Models/Project.java
package com.Techprimelab.Project.Models;

import java.sql.Date;
import java.util.List;

import org.springframework.stereotype.Component;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.PrimaryKeyJoinColumn;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Component
@Entity
@Table(name = "project")
public class Project {
	 	@Id
	    @GeneratedValue(strategy = GenerationType.IDENTITY)
	    @Column(name = "Id")
	    private int id;

	    @Column(name = "project_name")
	    private String projectName;

	    @ManyToOne
	    @JoinColumn(name = "reason_id", referencedColumnName = "reasonid")
	    private Reason reason;

	    @ManyToOne
	    @JoinColumn(name = "type_id", referencedColumnName = "typeid")
	    private Type type;

	    @ManyToOne
	    @JoinColumn(name = "division_id", referencedColumnName = "divisionid")
	    private Division division;

	    @ManyToOne
	    @JoinColumn(name = "category_id", referencedColumnName = "categoryid")
	    private Category category;

	    @ManyToOne
	    @JoinColumn(name = "priority_id", referencedColumnName = "priorityid")
	    private Priority priority;
	    @ManyToOne
	    @JoinColumn(name = "department_id", referencedColumnName = "departmentid")
	    private Department department;

	    @Column(name = "start_date")
	    private Date startDate;

	    @Column(name = "end_date")
	    private Date endDate;

	    @ManyToOne
	    @JoinColumn(name = "location_id", referencedColumnName = "locationid")
	    private Location location;

	    @ManyToOne
	    @JoinColumn(name = "status_id", referencedColumnName = "statusid")
	    private Status status;

	}

File: Project/src/main/app/models/project.ts
import { Category } from "./Category";
import { Department } from "./Department";
import { Division } from "./Division";
import { Priority } from "./Priority";
import { Reasons } from "./Reasons";
import { Type } from "./Type";
import { Location } from "./Location";
import { Status } from "./status";
import { User } from "./user";

export class Project {
  id: number;
  projectName: string;
  reason: Reasons;
  type: Type;
  division: Division;
  category: Category;
  priority: Priority;
  department: Department;
  startDate: Date;
  endDate: Date;
  location: Location;
  status: Status;
  user: User;
  

 
  constructor(
    id: number = 0,
    projectName: string = '',
    reason: Reasons = new Reasons(1,""),
    type: Type = new Type(1,""),
    division: Division = new Division(1,""),
    category: Category = new Category(1,""),
    priority: Priority = new Priority(1,""),
    department: Department = new Department(1,""),
    startDate: Date = new Date(),
    endDate: Date = new Date(),
    location: Location = new Location(1,""),
    status: Status = new Status(1,""),
    user: User = new User()
  ) {
    this.id = id;
    this.projectName = projectName;
    this.reason = reason;
    this.type = type;
    this.division = division;
    this.category = category;
    this.priority = priority;
    this.department = department;
    this.startDate = startDate;
    this.endDate = endDate;
    this.location = location;
    this.status = status;
    this.user = user;
  }
}